packages/shopping-cart/src/action-promise-error.ts

```diff
@@ -0,0 +1,9 @@
+export class ActionPromiseError extends Error {
+	code: string;
+
+	constructor( message: string, code: string ) {
+		super( message );
+		Object.setPrototypeOf( this, ActionPromiseError.prototype );
```

Comment:
Why do you need to call `setPrototypeOf`? Using `class` should automatically set up all the prototypes and chains etc. Is there something special about `Error` in particular?

Reply:
I don't fully understand this, but it was recommended by [the TypeScript docs](https://github.com/Microsoft/TypeScript-wiki/blob/main/Breaking-Changes.md#user-content-extending-built-ins-like-error-array-and-map-may-no-longer-work).

Reply:
I understand less than half of it 🙂 but apparently the `Error` class is weird in many ways and extending it is more difficult than other classes.